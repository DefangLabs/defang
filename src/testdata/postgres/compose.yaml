services:
  with-ext:
    image: postgres
    x-defang-postgres:
      allow-downtime: true
    ports:
      - target: 5432
        mode: host

  wrong-image:
    image: example
    x-defang-postgres:
    ports:
      - target: 5432
        mode: host

  short-ports:
    image: postgres
    x-defang-postgres:
    ports:
      - 5432:5432

  no-ext:
    image: postgres
    ports:
      - target: 5432
        mode: host

  no-ports:
    image: postgres
    x-defang-postgres:

  no-ports-override:
    image: postgres
    x-defang-postgres:
    environment:
      - PGPORT=5433

