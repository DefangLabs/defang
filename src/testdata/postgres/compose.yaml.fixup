{
  "no-ext": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "postgres",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 5432,
        "protocol": "tcp"
      }
    ]
  },
  "no-ports": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "postgres",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 5432,
        "protocol": "tcp"
      }
    ]
  },
  "no-ports-override": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "environment": {
      "PGPORT": "5433"
    },
    "image": "postgres",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 5433,
        "protocol": "tcp"
      }
    ]
  },
  "with-ext": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "postgres",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 5432,
        "protocol": "tcp"
      }
    ]
  },
  "wrong-image": {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "example",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 5432,
        "protocol": "tcp"
      }
    ]
  }
}