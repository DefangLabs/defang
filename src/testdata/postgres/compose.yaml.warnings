 ! service "no-ext": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "no-ext": stateful service will lose data on restart; use a managed service instead
 ! service "no-ports": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "no-ports-override": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "with-ext": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "wrong-image": managed Postgres service should use a postgres image
 ! service "wrong-image": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors