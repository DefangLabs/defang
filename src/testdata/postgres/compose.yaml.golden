name: postgres
services:
  no-ext:
    image: postgres
    networks:
      default: null
    ports:
      - mode: host
        target: 5432
        protocol: tcp
  no-ports:
    image: postgres
    networks:
      default: null
    x-defang-postgres: null
  no-ports-override:
    environment:
      PGPORT: "5433"
    image: postgres
    networks:
      default: null
    x-defang-postgres: null
  with-ext:
    image: postgres
    networks:
      default: null
    ports:
      - mode: host
        target: 5432
        protocol: tcp
    x-defang-postgres:
      allow-downtime: true
  wrong-image:
    image: example
    networks:
      default: null
    ports:
      - mode: host
        target: 5432
        protocol: tcp
    x-defang-postgres: null
networks:
  default:
    name: postgres_default
